sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallAutomationClientAutomatedLiveTestsBase.cs

```diff
@@ -0,0 +1,249 @@
+
+        protected async Task<CommunicationUserIdentifier> CreateIdentityUserAsync()
+        {
+            CommunicationIdentityClient communicationIdentityClient = CreateInstrumentedCommunicationIdentityClient();
+            return await communicationIdentityClient.CreateUserAsync().ConfigureAwait(false);
+        }
+
+        private async Task MessageHandler(ProcessMessageEventArgs args)
+        {
+            string body = args.Message.Body.ToString();
+
+            if (!string.IsNullOrEmpty(body))
+            {
+                Console.WriteLine($"Events: {JToken.Parse(body).ToString(Formatting.Indented)}");
+
+                if (body.Contains("\"incomingCallContext\":\""))
+                {
+                    // for incoming call event grid event
+                    string incomingCallContext = body.Split(new[] { "\"incomingCallContext\":\"" }, StringSplitOptions.None)[1].Split(new[] { "\"" }, StringSplitOptions.None)[0];
+
+                    string uniqueId = ParseFromAndTo(body);
+
+                    _incomingcontextstore.TryAdd(RemoveAllNonChar(uniqueId), incomingCallContext);
+                }
+                else
+                {
+                    // for call automation callback events
+                    CallAutomationEventBase callBackEvent = CallAutomationEventParser.Parse(BinaryData.FromString(body));
+
+                    if (_eventstore.TryGetValue(callBackEvent.CallConnectionId, out var mylist))
+                    {
+                        mylist.AddOrUpdate(callBackEvent.GetType(), callBackEvent, (key, oldValue) => callBackEvent);
+                    }
+                    else
+                    {
+                        ConcurrentDictionary<Type, CallAutomationEventBase> events = new ConcurrentDictionary<Type, CallAutomationEventBase>();
+                        events.TryAdd(callBackEvent.GetType(), callBackEvent);
+                        _eventstore.Add(callBackEvent.CallConnectionId, events);
+                    }
+                }
+            }
+
+            // complete the message. messages is deleted from the queue.
+            await args.CompleteMessageAsync(args.Message);
+        }
+
+        private Task ErrorHandler(ProcessErrorEventArgs args)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="CallAutomationClientOptions" />
+        /// </summary>
+        /// <returns>The instrumented <see cref="CallAutomationClientOptions" />.</returns>
+        private CallAutomationClientOptions CreateServerCallingClientOptionsWithCorrelationVectorLogs()
+        {
+            CallAutomationClientOptions callClientOptions = new CallAutomationClientOptions();
+            callClientOptions.Diagnostics.LoggedHeaderNames.Add("MS-CV");
+            return InstrumentClientOptions(callClientOptions);
+        }
+
+        private string ParseFromAndTo(string unparsed)
```

Comment:
nit: is this the Unique Id? would it be more clear to name the function ParseUniqueId?